### PR TITLE
Add virsh command support and Add a new object in nwfilter_xml

### PR DESCRIPTION
**Pull request by juteman**
virsh command: add 4 virsh command support
- nwfilter_binding (create, delete, list, dumpxml)

add object: add object for nwfilter binding create params
xml format

Signed-off-by: qiulzhan <qiulzhan@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
Looks good to me

File: virttest/libvirt_xml/nwfilter_xml.py [hunk 1]
```diff
@@ -384,3 +383,131 @@ def get_all_protocols(self, protocol=None):
...
 
         return protocols
+
+
+class NwfilterBinding(base.LibvirtXMLBase):
+    """
+    Accessor methods for NwfilterXML class
+
+    Properties:
+        owner:
```
Comment (chunfuwen) [inline virttest/libvirt_xml/nwfilter_xml.py]: suggest add simple NwfilterXML template example here

Comment (juteman) [reply virttest/libvirt_xml/nwfilter_xml.py]:
```
<filterbinding>
  <owner>
    <name>rhel</name>
<uuid>c9543ff4-1703-40a9-abbf-7f3e0fcd66c0</uuid>
  </owner>
  <portdev name='vnet0'/>
<mac address='52:54:00:fa:81:87'/>
  <filterref filter='clean-traffic'>
<parameter name='MAC' value='52:54:00:fa:81:87'/>
  </filterref>
</filterbinding>
```

and i have separate it to a new file. please check

Comment (juteman) [reply virttest/libvirt_xml/nwfilter_xml.py]: i have separate the object to a file. And add the simple example

File: virttest/libvirt_xml/nwfilter_xml.py [hunk 2]
```diff
@@ -127,7 +127,6 @@ def del_protocol(self):
 
 
 class NwfilterXMLBase(base.LibvirtXMLBase):
```
Comment (chunfuwen) [inline virttest/libvirt_xml/nwfilter_xml.py]: I don't think you need changes in this python file

File: virttest/virsh.py [hunk 3]
```diff
@@ -3856,6 +3856,65 @@ def nwfilter_edit(name, options="", **dargs):
     return command(cmd, **dargs)
 
 
+def nwfilter_binding_create(name, options="", **dargs):
```
Comment (chunfuwen) [inline virttest/virsh.py]: function naming convention: verb + n.

File: virttest/virsh.py [hunk 4]
```diff
@@ -3856,6 +3856,65 @@ def nwfilter_edit(name, options="", **dargs):
...
+    :param name: binding xml file name
+    :param options: extra options to nwfilter-binding- cmd.
+    :param dargs: standardized virsh function API keywords
+    :return: CmdResult object
+    """
+    cmd = "nwfilter-binding-create %s %s" % (name, options)
+    return command(cmd, **dargs)
+
+
+def nwfilter_binding_list(options="", **dargs):
```
Comment (chunfuwen) [inline virttest/virsh.py]: same issue as before

File: virttest/libvirt_xml/nwfilter_binding.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+        name:
+        string. filter name
+        parameters:
+        list. parameters element dict list
+        """
+        __slots__ = ("name", "parameters")
+
+        def __init__(self, virsh_instance=base.virsh):
+            accessors.XMLAttribute(property_name="name",
```
Comment (chunfuwen) [inline virttest/libvirt_xml/nwfilter_binding.py]: Better keep property_name same with attribute

File: virttest/virsh.py (hunk 3 above)
Comment (juteman) [reply virttest/virsh.py]: the virsh command is
```
nwfilter-binding-create xmlfile
```
so to make it the same as virsh command. 
its: n+verb

File: virttest/libvirt_xml/nwfilter_binding.py (hunk 5 above)
Comment (juteman) [reply virttest/libvirt_xml/nwfilter_binding.py]: thks, i change the interface.py to make it scalable

**Comment by juteman** [comment]
run test
```
sudo avocado run --vt-type libvirt type_specific.io-github-autotest-libvirt.nwfilter_binding_list.variable_notation
JOB ID     : 6f81d0798bf6411a1ef1433dd9118348b3bcd617
JOB LOG    : /root/avocado/job-results/job-2018-08-10T14.01-6f81d07/job.log
 (1/1) type_specific.io-github-autotest-libvirt.nwfilter_binding_list.variable_notation: PASS (46.92 s)

```

**Comment by juteman** [comment]
and old case
```
sudo avocado run --vt-type libvirt type_specific.io-github-autotest-libvirt.nwfilter_edit_uuid
JOB ID     : 2332c4e156d33468cac74b4ad4a4e1d9dc77f097
JOB LOG    : /root/avocado/job-results/job-2018-08-10T14.03-2332c4e/job.log
 (1/1) type_specific.io-github-autotest-libvirt.nwfilter_edit_uuid: PASS (25.67 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 36.69 s

```

**Pull request merged**